wallet/src/actors/app/methods.rs

```diff
@@ -771,20 +783,29 @@ impl App {
             .flatten()
             .inspect(|res| {
                 log::debug!("Periodic request result: {:?}", res);
+                let status = serde_json::from_value::<types::SyncStatus>(res.clone());
+                log::debug!("The result of the node status is {:?}", status.unwrap());
             })
-            .map_err(|err| {
+            .into_actor(self)
+            .map_err(move |err, act, _ctx| {
                 log::warn!("Periodic request failed: {}", &err);
-                // TODO: detect node disconnected
-            })
-            .map(|_res| {
-                // Ignore result
+                log::info!("The node is disconnected");
```

Comment:
Can we do this only when the error is a timeout error?

Reply:
I've checked the possible errors, and I think the node can be considered disconnected as well.  The possible errors when sending a request are:
- `InvalidUrl`: The wallet is pointing an invalid url, so it OK to consider the node is disconnected. The error is handled during the wallet subscriptions.
- RequestTimedOut: The node is not responding, we can consider the node is disconnected.
-RequestFailed: this is due to a connection error or an event loop that is down, since in this request there is no event loop, we can assume it is given by a connection error and so it is OK to send the NodeDisconnected event.
-SerializeFailed: since this request has no params this error is not valid in this case.

Reply:
Lets log the error at least then, as log::error

Reply:
as we spoke, this is already being done. Nothing else then!